### DISCUSSION TITLE
자바의 static

### DISCUSSION THREAD

**Discussion by giscus** [stoneHee99/tistory_comment#4] [Comments]
# 자바의 static

자바를 사용해 코딩을 하다보면 static 이라는 키워드를 보게 됩니다!대표적으로 public static void main() 과 같은 구문에서도 볼 수 있죠! 오늘은 이 static 키워드가 무엇을 의미하고 어떻게 동작하는지 알아보겠습니다! 자바의 static자바에서 static 키워드는 클래스 수준에서 변수를 선언하거나 혹은 메서드를 정의할 때 사용합니다.static의 사용을 통해 클래스의 인스턴스에 속하는 것이 아닌 클래스 자체에 속하는 멤버를 만들 수 있습니다.예를 들어볼까요?static 변수static 변수는 정적 변수 혹은 클래스 변수라고도 합니다. 클래스의 모든 인스턴스가 공유하는 변수 입니다.public class Counter {    private static int count = 0..

https://stonehee99.tistory.com/35

**Comment by yoonseon12** [comment]
평소에 OOP에 대해 고민하면서 코드를 작성하고 있는데, `static`을 남발하면 객체 지향을 위배할 수 있다는 내용이 흥미롭고 좋은 인사이트 였습니다.
좋은 글 감사합니다!!
